File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> freq(26, 0);
        queue<pair<char, int>> q;  // store (char, index)

        // Count frequency and track order
        for (int i = 0; i < s.size(); i++) {
            freq[s[i] - 'a']++;
            q.push({s[i], i});
        }

        // Remove all repeated chars from front of queue
        while (!q.empty() && freq[q.front().first - 'a'] > 1) {
            q.pop();
        }

        return q.empty() ? -1 : q.front().second;
    }
};

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        Ns=s.lower()
        str_as_arr=list(Ns)
        for i in range(len(str_as_arr)):
            not_dup=True
            for j in range ( len(str_as_arr)):
                if i!=j and str_as_arr[i]==str_as_arr[j]:
                    not_dup=False
                    break
            if not_dup:
                return i
    
        return -1
        